crates/forge_services/src/provider_registry.rs

```diff
@@ -0,0 +1,22 @@
+use std::sync::Arc;
+
+use forge_app::ProviderRegistry;
+use forge_domain::{ForgeKey, Provider};
+
+use crate::ProviderInfra;
+
+pub struct ForgeProviderRegistry<F> {
+    infra: Arc<F>,
+}
+
+impl<F: ProviderInfra> ForgeProviderRegistry<F> {
+    pub fn new(infra: Arc<F>) -> Self {
+        Self { infra }
+    }
+}
+
+impl<F: ProviderInfra> ProviderRegistry for ForgeProviderRegistry<F> {
```

Comment:
Registry is responsible for getting the provider. Logical Flow:

```rust
trait ProviderRegistry {
  fn resolve(&self, config: &ForgeConfig) -> anyhow::Result<Provider>;
}
```

- Check if config contains provider information, then create a provider from it and return.
- Otherwise use the `EnvInfra` to read different variables in an order of preference and return a Provider.